src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -10,29 +10,103 @@ trait SortedSets {
 
   final val zadd = Command(
     "ZADD",
-    Tuple6(
+    Tuple5(
       StringInput,
       OptionalInput(ZAddUpdateInput),
       OptionalInput(ZAddChangedInput),
       OptionalInput(ZAddIncrementInput),
-      LongInput,
-      NonEmptyList(StringInput)
+      NonEmptyList(ZAddMemberScoresInput)
     ),
     LongOutput
   )
 
   final val zcard   = Command("ZCARD", StringInput, LongOutput)
   final val zcount  = Command("ZCOUNT", Tuple2(StringInput, RangeInput), LongOutput)
   final val zincrby = Command("ZINCRBY", Tuple3(StringInput, LongInput, ByteInput), LongOutput)
+
+  final val zlexcount = Command("ZLEXCOUNT", Tuple2(StringInput, ZLexCountInput), LongOutput)
+
+  final val zinterstore = Command(
+    "ZINTERSTORE",
+    Tuple3(
+      StringInput,
+      NonEmptyList(ZInterstoreKey),
+      OptionalInput(ZInterstoreAggregate)
+    ),
+    LongOutput
+  )
+
+  final val zpopmax = Command("ZPOPMAX", Tuple2(StringInput, OptionalInput(LongInput)), StreamOutput)
+  final val zpopmin = Command("ZPOPMIN", Tuple2(StringInput, OptionalInput(LongInput)), StreamOutput)
+
   final val zrange =
     Command("ZRANGE", Tuple3(StringInput, RangeInput, OptionalInput(ZRangeWithScoresInput)), StreamOutput)
-  final val zrank    = Command("ZRANK", Tuple2(StringInput, ByteInput), ByteOutput)
-  final val zrem     = Command("ZREM", Tuple2(StringInput, NonEmptyList(ByteInput)), LongOutput)
+
+  final val zrangebylex = Command(
+    "ZRANGEBYLEX",
+    Tuple3(
+      StringInput,
+      ZRangeByLexRangeInput,
+      OptionalInput(ZRangeByLexLimitInput)
+    ),
+    StreamOutput
+  )
+
+  final val zrangebyscore = Command(
+    "ZRANGEBYSCORE",
+    Tuple4(
+      StringInput,
+      ZRangeByScoreRangeInput,
+      OptionalInput(ZRangeWithScoresInput),
```

Comment:
Regarding the comment in PR description: please make `WithScores` a case object. That should suffice. Alternatively, you can have a helper lazy val with instantiated `WithScores`.

Reply:
I've used case objects for now.